Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM python:3.7-slim-stretch as base
+
+RUN apt-get update && \
+    apt-get install --yes curl netcat
+
+RUN pip3 install --upgrade pip
+RUN pip3 install virtualenv
+
+RUN virtualenv -p python3 /appenv
+
+RUN . /appenv/bin/activate; pip install -U pip
```

Comment:
Discovered a nicer way of doing this recently. Rather than activating the virtualenv before every python command, you can simply prepend it to the path:

```
ENV PATH=/appenv/bin:$PATH
```

Reply:
Cool! Incorporated this suggestion.